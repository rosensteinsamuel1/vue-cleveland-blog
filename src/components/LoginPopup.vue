<template>
  <div class="text-center">
    <v-dialog v-model="dialog" width="500">
      <v-card>
        <v-card-title class="headline grey lighten-2">Log In</v-card-title>

        <v-card-text>
          <v-form class="mt-5 px-3" ref="form">
            <v-text-field v-model="email" label="Email" :rules="inputRules"></v-text-field>
            <v-text-field v-model="password" type="password" label="Password" :rules="inputRules"></v-text-field>
          </v-form>
        </v-card-text>

        <v-divider></v-divider>
        <v-card-text v-if="error">
          <p class="red--text">{{error}}</p>
        </v-card-text>

        <v-card-actions>
          <v-spacer></v-spacer>
          <v-btn color="primary mx-0 mt-3" text @click="close">Log In</v-btn>
        </v-card-actions>
      </v-card>
    </v-dialog>
  </div>
</template>

<script>
import { Auth } from "../firebase/auth";
import { USER_GET_INFO } from "../store/action-types";

export default {
  data() {
    return {
      dialog: false,
      email: "",
      password: "",
      username: "",
      error: "",
      inputRules: [v => v.length >= 3 || "Minimum length is 3 characters"]
    };
  },
  methods: {
    open() {
      this.dialog = true;
    },
    close() {
      if (this.$refs.form.validate()) {
        // Log in with existing user
        Auth.signInWithEmailAndPassword(this.email, this.password)
          .then(response => {
            console.log(response.user.uid);
            // retrieve user data from Firestore
            this.$store.dispatch(USER_GET_INFO, response.user.uid);
            this.error = null;
            this.dialog = false;
          })
          .catch(error => {
            this.error = `${error.message} [${error.code}]`;
          });
      }
    }
  }
};
</script>