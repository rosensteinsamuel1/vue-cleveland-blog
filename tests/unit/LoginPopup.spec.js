import Vue from 'vue'
import Vuex from 'vuex'
import vuetify from 'vuetify'
import componentStore from '@/store/store'

import { shallowMount } from '@vue/test-utils'
import LoginPopup from '@/components/LoginPopup'

Vue.use(Vuex)
Vue.use(vuetify)

const store = new Vuex.Store(componentStore)

describe('LoginPopup Component', () => {
    it('renders', () => {
        const wrapper = shallowMount(LoginPopup, {
            store
        })
        expect(wrapper.exists()).toBeTruthy()
        expect(wrapper.html()).toMatchSnapshot()
    })
    it('matches snapshot', () => {
        const wrapper = shallowMount(LoginPopup, {
            store
        })
        expect(wrapper).toMatchSnapshot()
    })
})