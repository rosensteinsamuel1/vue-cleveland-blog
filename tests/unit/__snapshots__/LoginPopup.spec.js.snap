// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPopup Component matches snapshot 1`] = `
<div class="text-center">
  <v-dialog-stub opendelay="0" closedelay="0" contentclass="" maxwidth="none" origin="center center" retainfocus="true" transition="dialog-transition" width="500">
    <v-card-stub loaderheight="4" tag="div">
      <v-card-title-stub class="headline grey lighten-2">Log In</v-card-title-stub>
      <v-card-text-stub>
        <v-form-stub class="mt-5 px-3">
          <v-text-field-stub errorcount="1" errormessages="" messages="" rules="function (v) {
        return v.length >= 3 || &quot;Minimum length is 3 characters&quot;;
      }" successmessages="" value="" backgroundcolor="" label="Email" loaderheight="2" clearicon="$clear" type="text"></v-text-field-stub>
          <v-text-field-stub errorcount="1" errormessages="" messages="" rules="function (v) {
        return v.length >= 3 || &quot;Minimum length is 3 characters&quot;;
      }" successmessages="" value="" backgroundcolor="" label="Password" loaderheight="2" clearicon="$clear" type="password"></v-text-field-stub>
        </v-form-stub>
      </v-card-text-stub>
      <v-divider-stub></v-divider-stub>
      <!---->
      <v-card-actions-stub>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub color="primary mx-0 mt-3" tag="button" activeclass="" text="true" type="button">Log In</v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog-stub>
</div>
`;

exports[`LoginPopup Component renders 1`] = `
<div class="text-center">
  <v-dialog-stub opendelay="0" closedelay="0" contentclass="" maxwidth="none" origin="center center" retainfocus="true" transition="dialog-transition" width="500">
    <v-card-stub loaderheight="4" tag="div">
      <v-card-title-stub class="headline grey lighten-2">Log In</v-card-title-stub>
      <v-card-text-stub>
        <v-form-stub class="mt-5 px-3">
          <v-text-field-stub errorcount="1" errormessages="" messages="" rules="function (v) {
        return v.length >= 3 || &quot;Minimum length is 3 characters&quot;;
      }" successmessages="" value="" backgroundcolor="" label="Email" loaderheight="2" clearicon="$clear" type="text"></v-text-field-stub>
          <v-text-field-stub errorcount="1" errormessages="" messages="" rules="function (v) {
        return v.length >= 3 || &quot;Minimum length is 3 characters&quot;;
      }" successmessages="" value="" backgroundcolor="" label="Password" loaderheight="2" clearicon="$clear" type="password"></v-text-field-stub>
        </v-form-stub>
      </v-card-text-stub>
      <v-divider-stub></v-divider-stub>
      <!---->
      <v-card-actions-stub>
        <v-spacer-stub></v-spacer-stub>
        <v-btn-stub color="primary mx-0 mt-3" tag="button" activeclass="" text="true" type="button">Log In</v-btn-stub>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog-stub>
</div>
`;
